### DISCUSSION TITLE
How to create specific sized windows for an i3 layout/workspace

### DISCUSSION THREAD

**Discussion by Yeet7** [i3/i3#6090] [Q&A]
Heyo, just wondering how I could create a setup (to run off of a keybind or something) that will set the windows I have there to specific sizes. Note, I have a g9, so its a 49" ultrawide, that's why. So e.g. chrome on the left, game in the middle (set to specifically 2650x1440) and discord on the right, and chrome & discord are the same size. But it can be anything in any of the positions, just in that specific layout (two on the ends are equal size and middle is 2650x1440).

I'm also not entirely sure how this would work with polybar and gaps? Just a side-question tbh, I'll figure it out.

Found an old reddit post while looking for answers but that's from 6 years ago and hasn't been updated in 2, so I doubt that'll work with all the updates since. There is also [this](https://github.com/i3/i3/discussions/5917) post from february, which looks promising, but it specifies to chrome, which partially leads me to believe it isn't possible to configure it with _any_ program in _any_ position, that it must be configured on a program-by-program basis essentially.

**Comment by KebabLord** [comment]
maybe layout saving is what you are looking for:
https://i3wm.org/docs/layout-saving.html

**Reply by Yeet7** [reply]
While this is very very helpful, I'm looking for something more permanent or like toggle-able. Where even if I close one or all windows, it will still take that that layout, or if I don't want it on, I can just turn it off. (Though the last part I can solve by just switching workspaces)

edit; 
Made a keybind out of it, which does seem to address the toggle-able issue, but I still don't quite understand how I could get it to work with 
A) any program
and
B) the programs already open, e.g. having chrome game and discord already open, just reformatting them.
